proposals/extensions_v2.md

```diff
@@ -0,0 +1,356 @@
+
+```c#
+var v1 = ((Extension)receiver).ExtensionMethod(); // instead of Extension.ExtensionMethod(receiver)
+var v2 = ((Extension)receiver).ExtensionProperty;
+var v3 = ((Extension)receiver)[indexerArg];
+var v4 = (Extension)receiver1 + receiver2;
+```
+
+Constructors and static methods would not need any special syntax as the extension can cleanly be referenced as a type where needed.
+
+```c#
+var v3 = new Extension(...); // Makes instance of the actual extended type.
+var v4 = Extension.StaticExtensionMethod(...);
+```
+
+Note 1: while the cast syntax traditionally casts or converts a value, that would not be the case for its use here.  It would only be used as a lookup mechanism to indicate which extension gets priority.  Importantly, even with this syntax, extensions themselves are not types.  For example:
+
+```c#
+Extension e1;                   // Not legal.  Extension is not a type.
+Extension[] e2;                 // Not legal.  Extension is not a type.
+List<Extension> e3;             // Not legal.  Extension is not a type.
+var v1 = (Extension)receiver;   // Not legal.  Can't can't have a value of extension type.
+```
+
+This is exactly the same as the restrictions on static-types *except* with the carve out that you can use the extension in a cast-syntax or new-expression *only* for lookup purposes and nothing else.
+
+Note 2. If cast syntax is not desirable here (especially if confuses the idea if extensions are types), we can come up with a new syntactic form.  We are not beholden to the above syntax.
+
+# Future expansion
+
+The above initial strawman solves several major goals for we want for the extensions space:
+
+1. Supporting a much broader set of extension member types.
+2. Having a clean syntax for extension members that matches the regular syntax form (in other words, an extension proeprty still looks like a property).
+3. Ensuring teams can move safely to modern extensions *especially* in environments where source binary compatibility is non-negotiable.
+
+However, there are parts of its core design that are not ideal which we would like to ensure we can expand on.  These expansions could be released with extensions if time and other resources permit.  Or they could came later and cleanly sit on top of the feature to improve the experience.
+
+These areas are:
+
+## Expansion 1: Syntactic clumsiness and repetition
+
+The initial extension form values source and binary compatibility as core requirements that must be present to ensure easy migration, allowing codebases to avoid both:
+1. bifurcation; where some codebases adopt modern extensions and some do not.
+2. internal inconsistency; where some codebases must keep around old extensions and new extensions, with confusion about the semantics of how each interacts with the other.
+
+Because classic extension methods have very few restrictions, modern extension methods need to be flexible enough to support all the scenarios supported there.
```

Comment:
```suggestion
Because classic extension methods have very few restrictions, modern extension methods need to be flexible enough to support all the scenarios which they support.
```